index.ts

```diff
@@ -135,7 +135,7 @@ export class PythonShell extends EventEmitter{
         this.mode = options.mode || 'text';
         this.formatter = resolve('format', options.formatter || this.mode);
         this.parser = resolve('parse', options.parser || this.mode);
-        this.stderrParser = resolve('parse', options.stderrParser || this.mode);
+        this.stderrParser = resolve('parse', options.stderrParser || 'text');
```

Comment:
```suggestion
        // We don't expect users to ever format stderr as JSON so we default to text mode
        this.stderrParser = resolve('parse', options.stderrParser || 'text');
```

Reply:
More comments the better IMO. The PR also needs a documentation update but I can do that myself after I merge in the PR :)